common/lib/xmodule/xmodule/tests/test_word_cloud.py

```diff
@@ -38,15 +38,15 @@ def test_good_ajax_request(self):
             response['student_words'],
             {'sun': 1, 'dog': 6, 'cat': 12}
         )
-        self.assertListEqual(
-            response['top_words'],
-            [{'text': 'dad', 'size': 2, 'percent': 9.0},
-             {'text': 'sun', 'size': 1, 'percent': 5.0},
-             {'text': 'dog', 'size': 6, 'percent': 27.0},
-             {'text': 'mom', 'size': 1, 'percent': 5.0},
-             {'text': 'cat', 'size': 12, 'percent': 54.0}]
-        )
 
-        self.assertEqual(
-            100.0,
-            sum(i['percent'] for i in response['top_words']))
+        expected_top_words = [{'text': 'dad', 'size': 2, 'percent': 9},
+                              {'text': 'sun', 'size': 1, 'percent': 5},
+                              {'text': 'dog', 'size': 6, 'percent': 27},
+                              {'text': 'mom', 'size': 1, 'percent': 5},
+                              {'text': 'cat', 'size': 12, 'percent': 54}]
+
+        for item in response['top_words']:
+            self.assertDictEqual(item, next(
+                item for expected_item in expected_top_words if expected_item["text"] == item["text"]))
```

Comment:
Isn't this just testing item against itself,  I think to get the right expected_item, you would want `expected_item for expected_item in expected_top_words if expected_item["text"] == item["text"]`

But since you're re-building the generator each time, I would expect the `next` call to not do what you want.

You may just want to have your expected_top_words be a dict from the tex to the dict that we expect for that text.  That way you can just look up the right dict for each item in the response and do the comparison.

Reply:
good catch @feanil